library/test/src/types.rs

```diff
@@ -140,6 +144,34 @@ impl TestDesc {
             }
         }
     }
+
+    #[cfg(not(bootstrap))]
+    pub fn test_mode(&self) -> &'static str {
+        if self.ignore {
+            return &"";
```

Comment:
A string literal `("...")` is already a `&str`; no need to add another `&`.

```suggestion
            return "";
```
(Same for all the others below.)

Reply:
It might be better to return a `Option<&'static str>` and use `None` instead of `""` as the special value.